receiver/gitproviderreceiver/internal/scraper/githubscraper/helpers.go

```diff
@@ -0,0 +1,179 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package githubscraper // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/gitproviderreceiver/internal/scraper/githubscraper"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/Khan/genqlient/graphql"
+	"github.com/google/go-github/v53/github"
+	"go.uber.org/zap"
+)
+
+func (ghs *githubScraper) getRepos(
+	ctx context.Context,
+	client graphql.Client,
+	searchQuery string,
+) ([]SearchNodeRepository, int, error) {
+	// here we use a pointer to a string so that graphql will receive null if the
+	// value is not set since the after: $repoCursor is optional to graphql
+	var cursor *string
+	var repos []SearchNodeRepository
+	var count int
+
+	for next := true; next; {
+		r, err := getRepoDataBySearch(ctx, client, searchQuery, cursor)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting repo data", zap.Error(err))
+			return nil, 0, err
+		}
+
+		for _, repo := range r.Search.Nodes {
+			if r, ok := repo.(*SearchNodeRepository); ok {
+				repos = append(repos, *r)
+			}
+		}
+
+		count = r.Search.RepositoryCount
+		cursor = &r.Search.PageInfo.EndCursor
+		next = r.Search.PageInfo.HasNextPage
+	}
+
+	return repos, count, nil
+}
+
+func (ghs *githubScraper) getBranches(
+	ctx context.Context,
+	client graphql.Client,
+	repoName string,
+	defaultBranch string,
+) (int, error) {
+	var cursor *string
+	var count int
+
+	for next := true; next; {
+		r, err := getBranchData(ctx, client, repoName, ghs.cfg.GitHubOrg, 50, defaultBranch, cursor)
+		if err != nil {
+			ghs.logger.Sugar().Errorf("error getting branch data", zap.Error(err))
+			return 0, err
+		}
+		count = r.Repository.Refs.TotalCount
+		cursor = &r.Repository.Refs.PageInfo.EndCursor
+		next = r.Repository.Refs.PageInfo.HasNextPage
+	}
+	return count, nil
+}
+
+// Ensure that the type of owner is user or organization
+func checkOwnerTypeValid(ownertype string) (bool, error) {
+	if ownertype == "org" || ownertype == "user" {
+		return true, nil
+	}
+	return false, errors.New("ownertype must be either org or user")
+}
+
+// Check to ensure that the login user (org name or user id) exists or
+// can be logged into.
+func (ghs *githubScraper) checkOwnerExists(ctx context.Context, client graphql.Client, owner string) (exists bool, ownerType string, err error) {
```

Comment:
the code feels for me a bit too clever with named return parameters, but just a nit.

Reply:
Appreciate the callout. I generally default to named returns due to readability through the LSP. I'd be happy to simplify down though. Realistically, i don't even need to return a bool and can just return type & error where error takes the place of the bool.

Reply:
In hindsite, this function is a bit redundent. GitHub is only going to return org or user and all we really eed to know is that you can login, then adjust the search query based on type. I'm just going to remove this function and check entirely.

Reply:
There was another function that I deleted entirely though, the checkOwnerTypeValid function was the redundant function I was originally referring to.